In [1]:
from LyricsAnnot import LyricsAnnot
import json
import main
import utils

In [2]:
save_path = "./test/saved/"
id_file_path = "./test/test_id.txt"

# Test with a song from DAMP dataset

## Lyrics Annotation Class

In [3]:
# Read the data from a JSON file
file_path = './test/test_datasets/damp/ES/ESLyrics/3364824_3364824.json'

with open(file_path) as file:
    data = json.load(file)

In [4]:
# Create the Lyrics object
lyrics = LyricsAnnot('Perro Fiel', 'Shakira')

# Print its information
print(lyrics)

Searching for "Perro Fiel" by Shakira...


Done.
META-DATA
------------------------
ID:	None
Title:	Perro Fiel
Artist:	Shakira
Lang:	es
Writer Artists:	['Juan Diego Medina', 'Saga WhiteBlack', 'Nicky Jam', 'Shakira']
Duration:	195.213

ANNOTATIONS
------------------------
[]


In [5]:
# Build the annotations from the read data and print again
lyrics.build_annotations(data, 'DAMP')

print(lyrics)

META-DATA
------------------------
ID:	None
Title:	Perro Fiel
Artist:	Shakira
Lang:	es
Writer Artists:	['Juan Diego Medina', 'Saga WhiteBlack', 'Nicky Jam', 'Shakira']
Duration:	195.213

ANNOTATIONS
------------------------
[{'line': 'Aquí estás',
  'time_duration': 2.25,
  'time_index': [4.919791666666667, 7.069791666666667]},
 {'line': 'Ya no puedes detenerte',
  'time_duration': 2.5,
  'time_index': [7.169791666666667, 9.569791666666667]},
 {'line': '¿Dónde vas? ahh',
  'time_duration': 2.75,
  'time_index': [9.669791666666667, 12.319791666666667]},
 {'line': 'Si estoy loco por tenerte',
  'time_duration': 3.25,
  'time_index': [12.419791666666667, 15.569791666666667]},
 {'line': 'Cómo lo iba a saber eh ehh eh ehh',
  'time_duration': 5.180208333333335,
  'time_index': [15.669791666666667, 20.75]},
 {'line': 'Que te vería otra vez eh ehh eh ehh',
  'time_duration': 0.0,
  'time_index': [20.85, 20.75]},
 {'line': '',
  'time_duration': 6.360416666666666,
  'time_index': [20.85, 27.11

In [6]:
# Save it to a JSON file
lyrics.save_to_json(save_path, id_file_path, 'DAMP')

## Add information to the annotations related to lyrics section

In [7]:
lyrics.add_section_info()

Searching for "Perro Fiel" by Shakira...
Done.


True

In [8]:
print(lyrics)

META-DATA
------------------------
ID:	00000000
Title:	Perro Fiel
Artist:	Shakira
Lang:	es
Writer Artists:	['Juan Diego Medina', 'Saga WhiteBlack', 'Nicky Jam', 'Shakira']
Duration:	195.213

ANNOTATIONS
------------------------
[{'lines': [{'line': 'Aquí estás',
             'time_duration': 2.1500000000000004,
             'time_index': [4.919791666666667, 7.069791666666667]},
            {'line': 'Ya no puedes detenerte',
             'time_duration': 2.4000000000000004,
             'time_index': [7.169791666666667, 9.569791666666667]},
            {'line': '¿Dónde vas? ahh',
             'time_duration': 2.6500000000000004,
             'time_index': [9.669791666666667, 12.319791666666667]},
            {'line': 'Si estoy loco por tenerte',
             'time_duration': 3.1500000000000004,
             'time_index': [12.419791666666667, 15.569791666666667]},
            {'line': 'Cómo lo iba a saber eh ehh eh ehh',
             'time_duration': 5.080208333333333,
             'time

In [9]:
lyrics.save_to_json(save_path, id_file_path, 'DAMP')

# Test with a song from DALI dataset

## Lyrics Annotation Class

In [10]:
# Read the data from a JSON file
file_path = './test/test_datasets/dali/my_annot_name_0082266932594913a73b3302355076e2.json'

with open(file_path) as file:
    data = json.load(file)

In [11]:
# Create the Lyrics object
lyrics = LyricsAnnot("Geschwisterliebe", "Die Ärzte")

# Print its information
print(lyrics)

Searching for "Geschwisterliebe" by Die Ärzte...
Done.
META-DATA
------------------------
ID:	None
Title:	Geschwisterliebe
Artist:	Die Ärzte
Lang:	de
Writer Artists:	['Farin Urlaub']
Duration:	155.093

ANNOTATIONS
------------------------
[]


In [12]:
# Build the annotations from the read data and print again
lyrics.build_annotations(data, 'DALI')

print(lyrics)

META-DATA
------------------------
ID:	None
Title:	Geschwisterliebe
Artist:	Die Ärzte
Lang:	de
Writer Artists:	['Farin Urlaub']
Duration:	155.093

ANNOTATIONS
------------------------
[{'line': 'wir haben zusammen im sandkasten gesessen',
  'time_duration': 5.9880239520958085,
  'time_index': [40.098, 46.08602395209581]},
 {'line': 'beim doktor spielen wollt ich nur dein fieber messen',
  'time_duration': 5.9880239520958085,
  'time_index': [48.08203193612774, 54.07005588822355]},
 {'line': 'jetzt bist du vierzehn und du bist soweit',
  'time_duration': 4.9900199600798345,
  'time_index': [56.066063872255484, 61.05608383233532]},
 {'line': 'wir warten keine ewigkeit-',
  'time_duration': 7.318695941450443,
  'time_index': [64.05009580838322, 71.36879174983366]},
 {'line': 'sind unsre eltern auch dagegen',
  'time_duration': 4.324683965402514,
  'time_index': [72.03412774451097, 76.35881170991348]},
 {'line': 'ich wurd dich gerne mal flach legen',
  'time_duration': 7.318695941450429,
 

In [13]:
# Save it to a JSON file
lyrics.save_to_json(save_path, id_file_path, 'DALI')

## Add information to the annotations related to lyrics section

In [14]:
lyrics.add_section_info()

Searching for "Geschwisterliebe" by Die Ärzte...
Done.


True

In [15]:
print(lyrics)

META-DATA
------------------------
ID:	00000005
Title:	Geschwisterliebe
Artist:	Die Ärzte
Lang:	de
Writer Artists:	['Farin Urlaub']
Duration:	155.093

ANNOTATIONS
------------------------
[{'lines': [{'line': 'wir haben zusammen im sandkasten gesessen',
             'time_duration': 5.9880239520958085,
             'time_index': [40.098, 46.08602395209581]},
            {'line': 'beim doktor spielen wollt ich nur dein fieber messen',
             'time_duration': 5.9880239520958085,
             'time_index': [48.08203193612774, 54.07005588822355]},
            {'line': 'jetzt bist du vierzehn und du bist soweit',
             'time_duration': 4.9900199600798345,
             'time_index': [56.066063872255484, 61.05608383233532]},
            {'line': 'wir warten keine ewigkeit-',
             'time_duration': 7.318695941450443,
             'time_index': [64.05009580838322, 71.36879174983366]},
            {'line': 'sind unsre eltern auch dagegen',
             'time_duration': 4.3246

In [16]:
lyrics.save_to_json(save_path, id_file_path, 'DALI')

# Translation test

So as to standardize the paragraph's names to enable future SPARQL queries, we created a dictionary used to translate the headers from each language used in our datasets to english. Here are the standard names we use: Intro, Verse, Pre-Chorus, Chorus, Post-Chorus, Bridge and Outro. Any paragraph that cannot get translated to one of these gets put aside.

## DALI

In [17]:
# Read the data from a JSON file
file_path = './test/test_datasets/dali/my_annot_name_27c19af62d9d440fb7cc2c14ab853a7d.json'

#turkish song: 0f63c15316874f04b7b918ea4147877d

with open(file_path, 'r', encoding='utf-8') as file:
    data = json.load(file)

In [18]:
# Create the Lyrics object
lyrics = LyricsAnnot('Allt På Ett Kort', 'Bob Hund')

# Print its information
print(lyrics)

Searching for "Allt På Ett Kort" by Bob Hund...
Done.
META-DATA
------------------------
ID:	None
Title:	Allt På Ett Kort
Artist:	Bob Hund
Lang:	sv
Writer Artists:	['\u200bbob hund']
Duration:	150.786

ANNOTATIONS
------------------------
[]


In [19]:
# Build the annotations from the read data and print again
lyrics.build_annotations(data, 'DALI')

print(lyrics)

META-DATA
------------------------
ID:	None
Title:	Allt På Ett Kort
Artist:	Bob Hund
Lang:	sv
Writer Artists:	['\u200bbob hund']
Duration:	150.786

ANNOTATIONS
------------------------
[{'line': 'och jag har glomt min forsta sommar',
  'time_duration': 4.799999999999997,
  'time_index': [44.2965, 49.0965]},
 {'line': 'och jag har glomt min forsta vinter med sno',
  'time_duration': 4.837499999999999,
  'time_index': [49.734, 54.5715]},
 {'line': 'men jag har kvar ett kort av min far',
  'time_duration': 2.5124999999999957,
  'time_index': [55.209, 57.7215]},
 {'line': 'jag har glomt min forsta fotboll',
  'time_duration': 4.275000000000006,
  'time_index': [58.059, 62.334]},
 {'line': "men det ar svart att sag' om det spelar nan roll",
  'time_duration': 5.25,
  'time_index': [63.384, 68.634]},
 {'line': 'jag har kvar ett kort av',
  'time_duration': 1.5,
  'time_index': [68.934, 70.434]},
 {'line': 'och jag har glomt min forsta cykel',
  'time_duration': 4.462500000000006,
  'time_ind

In [20]:
# Save it to a JSON file
lyrics.save_to_json(save_path, id_file_path, 'DALI')

In [21]:
lyrics = LyricsAnnot._genius_compiler.get_lyrics('Allt På Ett Kort', 'Bob Hund')

Searching for "Allt På Ett Kort" by Bob Hund...


Done.


In [22]:
paragraphs = LyricsAnnot._genius_compiler.split_by_section(lyrics, 'Bob Hund', 'sv')

In [23]:
print(paragraphs)

{('Verse', 1): {'content': 'Och jag har glömt min första sommar Och jag har glömt min första vinter med snö Men jag har kvar ett kort av far Och jag har glömt min första fotboll Men det är svårt att säga om det spelar nån roll Jag har kvar ett kort av min…', 'singer': ['Bob Hund']}, ('Verse', 2): {'content': 'Och jag har glömt min första cykel Men jag har kvar en minnesnyckel av guld Och jag har kvar ett kort av far Och jag minns vår första flygtur Och jag minns en gång en regnskur Jag har kvar ett kort av far', 'singer': ['Bob Hund']}, ('Outro', 1): {'content': 'Men jag har kvar ett kort av min farYou might also likeEmbed', 'singer': ['Bob Hund']}}


## DAMP

In [24]:
DAMP_dir = './test/test_datasets/damp/'
dir = 'ES'
metadata_file = "3364824_3364824.txt"

In [25]:
track_metadata = utils.read_arrangement_file(f"{DAMP_dir}{dir}/{dir}ArrangementMeta/{metadata_file}")

title = utils.parse_title(track_metadata['Arrangement title'])
artists = utils.parse_artist_names(track_metadata['Arrangement artist'])

file_path = f"{DAMP_dir}{dir}/{dir}Lyrics/{metadata_file.split('.')[0]}.json"

# Read JSON file
with open(file_path, encoding='utf-8') as file:
    data = json.load(file)

# Clean the lyrics
data = utils.clean_damp_json(data)

Skipping line: Arrangement tags: (does not contain ': ')
DETECTED FRONT INDEX: -1
DETECTED BACK INDEX 78


In [26]:
if utils.compute_avg_line_len(data) > 7.0:
    # Create annotations
    annot = LyricsAnnot(title, artists[0])
    print(annot)

    annot.build_annotations(data, 'DAMP')
    print(annot)

Searching for "Perro Fiel" by Shakira...
Done.
META-DATA
------------------------
ID:	None
Title:	Perro Fiel
Artist:	Shakira
Lang:	es
Writer Artists:	['Juan Diego Medina', 'Saga WhiteBlack', 'Nicky Jam', 'Shakira']
Duration:	195.213

ANNOTATIONS
------------------------
[]
META-DATA
------------------------
ID:	None
Title:	Perro Fiel
Artist:	Shakira
Lang:	es
Writer Artists:	['Juan Diego Medina', 'Saga WhiteBlack', 'Nicky Jam', 'Shakira']
Duration:	195.213

ANNOTATIONS
------------------------
[{'line': 'Aquí estás',
  'time_duration': 2.25,
  'time_index': [4.919791666666667, 7.069791666666667]},
 {'line': 'Ya no puedes detenerte',
  'time_duration': 2.5,
  'time_index': [7.169791666666667, 9.569791666666667]},
 {'line': '¿Dónde vas? ahh',
  'time_duration': 2.75,
  'time_index': [9.669791666666667, 12.319791666666667]},
 {'line': 'Si estoy loco por tenerte',
  'time_duration': 3.25,
  'time_index': [12.419791666666667, 15.569791666666667]},
 {'line': 'Cómo lo iba a saber eh ehh eh ehh

In [27]:
lyrics = LyricsAnnot('Perro Fiel', 'Shakira')

Searching for "Perro Fiel" by Shakira...
Done.


In [28]:
# Save it to a JSON file
lyrics.save_to_json(save_path, id_file_path, 'DAMP')

In [29]:
lyrics = LyricsAnnot._genius_compiler.get_lyrics(title, artists[0])

Searching for "Perro Fiel" by Shakira...


Done.


In [30]:
paragraphs = LyricsAnnot._genius_compiler.split_by_section(lyrics, artists[0], 'es')

In [31]:
print(paragraphs)

{('Verse', 1): {'content': 'Aquí estás Ya no puedes detenerte ¿Dónde vas? Si estoy loco por tenerte Cómo lo iba a saber Que te vería otra vez', 'singer': ['Shakira', 'Nicky Jam']}, ('Chorus', 1): {'content': 'Tú me confundes, no sé qué hacer Yo lo que quiero es pasarla bien Yo tengo miedo de que me guste Y que vaya a enloquecer Si eso pasa yo seguiré Contigo aquí como un perro fiel Yo tengo miedo de que me guste Y que vaya a enloquecer', 'singer': ['Shakira', 'Nicky Jam']}, ('Verse', 2): {'content': 'Te hablo en serio mai, no estoy jugando Tanto tiempo pasa y nada Estas ganas no me aguanto Y aunque tú me esquives, yo te sigo deseando Dicen que tú eres peligrosa No le hago caso a esas cosas Dime qué está pasando Me tienes como un loco, soy un loco enamorado, eh Quiero saber cuánto me vas a insistir Y hasta dónde llegarías por mí Siento mucho la espera Pero valdrá la pena cuando te esté besando De la manera que te mueves así Yo te lo juro me voy a derretir Tú sabes que soy buena Por más 

# Test of duplicates 
### When converting a DAMP file while the same song from DALI dataset has already been converted, there should not be any duplicate of id

Let's use Diamonds by Rihanna as a use case.

## DALI

In [3]:
# Read the data from a JSON file
file_path = './test/test_datasets/dali/my_annot_name_54d200fbe0424767acc0e414759f049d.json'

with open(file_path, 'r', encoding='utf-8') as file:
    data = json.load(file)

In [10]:
# Create the Lyrics object
lyrics = LyricsAnnot('Diamonds', 'Rihanna')

# Print its information
print(lyrics)

Searching for "Diamonds" by Rihanna...
Done.
META-DATA
------------------------
ID:	None
Title:	Diamonds
Artist:	Rihanna
Lang:	en
Writer Artists:	['Tor Erik Hermansen', 'Mikkel Storleer Eriksen', '\u200bbenny blanco', 'Sia']
Duration:	225.146

ANNOTATIONS
------------------------
[]


In [5]:
# Build the annotations from the read data and print again
lyrics.build_annotations(data, 'DALI')

True

In [6]:
# Save it to a JSON file
lyrics.save_to_json(save_path, id_file_path, 'DALI')

In [7]:
print(lyrics)

META-DATA
------------------------
ID:	00000001
Title:	Diamonds
Artist:	Rihanna
Lang:	en
Writer Artists:	['Tor Erik Hermansen', 'Mikkel Storleer Eriksen', '\u200bbenny blanco', 'Sia']
Duration:	225.146

ANNOTATIONS
------------------------
[{'line': 'shine bright like a diamond',
  'time_duration': 1.5896739130434787,
  'time_index': [1.304, 2.8936739130434788]},
 {'line': 'shine bright like a diamond',
  'time_duration': 1.5896739130434776,
  'time_index': [6.521391304347827, 8.111065217391305]},
 {'line': 'find light in the beautiful sea',
  'time_duration': 2.078804347826088,
  'time_index': [11.738782608695654, 13.817586956521742]},
 {'line': 'i choose to be happy',
  'time_duration': 1.875,
  'time_index': [14.02139130434783, 15.89639130434783]},
 {'line': 'you and i',
  'time_duration': 0.8967391304347849,
  'time_index': [16.63008695652174, 17.526826086956525]},
 {'line': 'you and i',
  'time_duration': 0.8967391304347849,
  'time_index': [18.01595652173913, 18.912695652173916]}

## DAMP

In [13]:
DAMP_dir = "./test/test_datasets/damp/"
dir = 'FR'
metadata_file = "738276_127259.txt"

In [14]:
track_metadata = utils.read_arrangement_file(f"{DAMP_dir}{dir}/{dir}ArrangementMeta/{metadata_file}")

title = utils.parse_title(track_metadata['Arrangement title'])
artists = utils.parse_artist_names(track_metadata['Arrangement artist'])

file_path = f"{DAMP_dir}{dir}/{dir}Lyrics/{metadata_file.split('.')[0]}.json"

# Read JSON file
with open(file_path, encoding='utf-8') as file:
    data = json.load(file)

# Clean the lyrics
data = utils.clean_damp_json(data)

DETECTED FRONT INDEX: 3
DETECTED BACK INDEX 69


In [16]:
if utils.compute_avg_line_len(data) > 7.0:
    # Create annotations
    annot = LyricsAnnot(title, artists[0])

    annot.build_annotations(data, 'DAMP')
    print(annot)

Searching for "Diamonds" by Rihanna...
Done.
META-DATA
------------------------
ID:	None
Title:	Diamonds
Artist:	Rihanna
Lang:	en
Writer Artists:	['Tor Erik Hermansen', 'Mikkel Storleer Eriksen', '\u200bbenny blanco', 'Sia']
Duration:	225.146

ANNOTATIONS
------------------------
[{'line': 'Find light in the beautiful sea',
  'time_duration': 2.260416666666666,
  'time_index': [11.189583333333333, 13.35]},
 {'line': 'I choose to be happy',
  'time_duration': 2.5197916666666664,
  'time_index': [13.45, 15.869791666666666]},
 {'line': 'You and I, you and I',
  'time_duration': 3.0302083333333343,
  'time_index': [15.969791666666666, 18.9]},
 {'line': 'We’re like diamonds in the sky',
  'time_duration': 0.0,
  'time_index': [19.0, 18.9]},
 {'line': '',
  'time_duration': 2.5302083333333343,
  'time_index': [19.0, 21.430208333333333]},
 {'line': 'You’re a shooting star I see',
  'time_duration': 2.2697916666666664,
  'time_index': [21.530208333333334, 23.7]},
 {'line': 'A vision of ecstasy

In [18]:
# Save it to a JSON file
annot.save_to_json(save_path, id_file_path, 'DAMP')

In [19]:
print(annot)

META-DATA
------------------------
ID:	00000001
Title:	Diamonds
Artist:	Rihanna
Lang:	en
Writer Artists:	['Tor Erik Hermansen', 'Mikkel Storleer Eriksen', '\u200bbenny blanco', 'Sia']
Duration:	225.146

ANNOTATIONS
------------------------
[{'line': 'Find light in the beautiful sea',
  'time_duration': 2.260416666666666,
  'time_index': [11.189583333333333, 13.35]},
 {'line': 'I choose to be happy',
  'time_duration': 2.5197916666666664,
  'time_index': [13.45, 15.869791666666666]},
 {'line': 'You and I, you and I',
  'time_duration': 3.0302083333333343,
  'time_index': [15.969791666666666, 18.9]},
 {'line': 'We’re like diamonds in the sky',
  'time_duration': 0.0,
  'time_index': [19.0, 18.9]},
 {'line': '',
  'time_duration': 2.5302083333333343,
  'time_index': [19.0, 21.430208333333333]},
 {'line': 'You’re a shooting star I see',
  'time_duration': 2.2697916666666664,
  'time_index': [21.530208333333334, 23.7]},
 {'line': 'A vision of ecstasy',
  'time_duration': 2.5197916666666664,

It works, the id created to generate the json from the DALI dataset's file is reused to generate the json from the DAMP dataset's one.
Moreover, the information regarding the second dataset the song comes from is added to the source field in the id text file.

# Test with an asian song that should be avoided

We will carry out this test with the song Hurry Xmas from L'Arc~en~Ciel. This song is in Japanese (wrongly listed as english in the DALI dataset), and it is written with occidental letters. Since Genius only uses asian characters for asian songs, we need to discard this song.<br>
<br>
/!\ Be careful, once you run this function once, the song gets processed: its dataset file should get moved to dali_already_converted folder and its id gets added in the dali_avoided_songs file. If you want to rerun this function, make sure to:<br>
1) erase the section of this song from the dali_avoided_songs file
2) move its dataset file once again in the original folder (in our case, dali/test/)

In [44]:
metadata_file = "my_annot_name_752c98e592864d75b818347825aabc03.json"
DALI_dir = "../dali/test/"
avoided_songs_file_path = "./test/test_dali_avoided_songs.txt"
already_converted_path = "./test/dali_already_converted/"

In [45]:
main.process_dali_file(metadata_file, DALI_dir, save_path, already_converted_path, id_file_path, avoided_songs_file_path, )

DETECTED FRONT INDEX: -1
DETECTED BACK INDEX 62
Searching for "Hurry Xmas" by L'Arc~en~Ciel...
Done.


'Success: my_annot_name_752c98e592864d75b818347825aabc03.json'

# Conversion of the whole DALI dataset

In [4]:
DALI_dir = "../dali/dali_json/"
id_file_path = "./dali_id.txt"
avoided_songs_file_path = "./dali_avoided_songs.txt"
already_converted_path = "../dali/dali_already_converted"
save_path = "./saved/"

In [6]:
main.create_dali_notations(DALI_dir, save_path, already_converted_path, id_file_path, avoided_songs_file_path)

Creating notations for the file my_annot_name_6d7f12f8d1224068a6f8aaf31a893791.json


  0%|          | 0/10 [00:00<?, ?it/s]

DETECTED FRONT INDEX: -1
DETECTED BACK INDEX 95
Searching for "Smells Like Teen Spirit" by Nirvana...
Done.
Searching for "Smells Like Teen Spirit" by Nirvana...
Done.


 10%|█         | 1/10 [00:11<01:39, 11.04s/it]


Success: my_annot_name_6d7f12f8d1224068a6f8aaf31a893791.json
Creating notations for the file my_annot_name_b1ae9051705d4314872b1a19a1507a26.json


  0%|          | 0/10 [00:00<?, ?it/s]

DETECTED FRONT INDEX: -1
DETECTED BACK INDEX 95
Searching for "Girls Just Want To Have Fun" by Cyndi Lauper...
Done.
Searching for "Girls Just Want To Have Fun" by Cyndi Lauper...


 10%|█         | 1/10 [00:10<01:38, 10.90s/it]


Done.
Success: my_annot_name_b1ae9051705d4314872b1a19a1507a26.json
Creating notations for the file my_annot_name_b78005c19ffa450aa1b9d1ec48c32710.json


  0%|          | 0/10 [00:00<?, ?it/s]

DETECTED FRONT INDEX: -1
DETECTED BACK INDEX 80
Searching for "Royals" by Lorde...
Done.
Searching for "Royals" by Lorde...


 10%|█         | 1/10 [00:04<00:44,  4.95s/it]

Done.
Success: my_annot_name_b78005c19ffa450aa1b9d1ec48c32710.json


 10%|█         | 1/10 [00:04<00:44,  4.95s/it]


Creating notations for the file my_annot_name_cc5cdff338b5407fa33efc9c5e44c9cf.json


  0%|          | 0/10 [00:00<?, ?it/s]

DETECTED FRONT INDEX: -1
DETECTED BACK INDEX 42
Searching for "Take Me Home, Country Roads" by John Denver...
Done.
Searching for "Take Me Home, Country Roads" by John Denver...


 10%|█         | 1/10 [00:12<01:48, 12.09s/it]


Done.
Success: my_annot_name_cc5cdff338b5407fa33efc9c5e44c9cf.json
Creating notations for the file my_annot_name_cdd7cdedccc441f79af56832d817e362.json


  0%|          | 0/10 [00:00<?, ?it/s]

DETECTED FRONT INDEX: -1
DETECTED BACK INDEX 96
Searching for "...Baby One More Time" by Britney Spears...
Done.
Searching for "...Baby One More Time" by Britney Spears...


 10%|█         | 1/10 [00:11<01:47, 12.00s/it]


Done.
Success: my_annot_name_cdd7cdedccc441f79af56832d817e362.json
Creating notations for the file my_annot_name_cedbbeb641184ebbae8f01a909ea1da7.json


  0%|          | 0/10 [00:00<?, ?it/s]

DETECTED FRONT INDEX: 6
DETECTED BACK INDEX 26
Searching for "Wonderful Tonight" by Eric Clapton...
Done.
Searching for "Wonderful Tonight" by Eric Clapton...


 10%|█         | 1/10 [00:06<00:55,  6.14s/it]


Done.
Success: my_annot_name_cedbbeb641184ebbae8f01a909ea1da7.json
Creating notations for the file my_annot_name_d2462019670b4f46bca8df24ce1e8132.json


  0%|          | 0/10 [00:00<?, ?it/s]

DETECTED FRONT INDEX: -1
DETECTED BACK INDEX 66
Searching for "Wrecking Ball" by Miley Cyrus...
Done.
Searching for "Wrecking Ball" by Miley Cyrus...


 10%|█         | 1/10 [00:06<01:01,  6.79s/it]


Done.
Success: my_annot_name_d2462019670b4f46bca8df24ce1e8132.json
Creating notations for the file my_annot_name_f1ead3dd60ba4395a5385cd7b8f091fd.json


  0%|          | 0/10 [00:00<?, ?it/s]

DETECTED FRONT INDEX: -1
DETECTED BACK INDEX 65
Searching for "El Scorcho" by Weezer...
Done.
Searching for "El Scorcho" by Weezer...


 10%|█         | 1/10 [00:06<00:57,  6.44s/it]


Done.
Success: my_annot_name_f1ead3dd60ba4395a5385cd7b8f091fd.json
Creating notations for the file my_annot_name_f9db03fa14fb4028b5e34d68364a024d.json


  0%|          | 0/10 [00:00<?, ?it/s]

DETECTED FRONT INDEX: -1
DETECTED BACK INDEX 153
Searching for "Say My Name" by Destiny's Child...
Done.
Searching for "Say My Name" by Destiny's Child...


 10%|█         | 1/10 [00:06<00:58,  6.54s/it]


Done.
Success: my_annot_name_f9db03fa14fb4028b5e34d68364a024d.json
Creating notations for the file my_annot_name_fc6d7838d422425680b3c529604f8df3.json


  0%|          | 0/10 [00:00<?, ?it/s]

DETECTED FRONT INDEX: -1
DETECTED BACK INDEX 48
Searching for "Burning Down The House" by Talking Heads...
Done.
Searching for "Burning Down The House" by Talking Heads...


 10%|█         | 1/10 [00:05<00:47,  5.31s/it]

Done.
Success: my_annot_name_fc6d7838d422425680b3c529604f8df3.json


## Statistics about the DALI dataset

In [7]:
utils.count_sources(id_file_path)

{'count_dali': 3364, 'count_damp': 0, 'count_double_sources': 0}

In [8]:
utils.calculate_percentage(id_file_path, already_converted_path, 'DALI')

'63.86% of the songs from DALI dataset are usable to create the feed the knowledge graph.'

In [9]:
utils.dali_count_avoided_songs(avoided_songs_file_path)

{'dali_counts': {'no_language_information': 28,
  'wrongly_encoded_asian_song': 48,
  'no_paragraphs': 1801,
  'not_found_on_Genius': 27,
  'total': 1904}}

# Conversion of the whole DAMP dataset

### Paths for test

In [2]:
fr_file = '738276_127259.txt'
fr_dir = "FR"

es_file = '738276_127259.txt'
es_dir = "US"

DAMP_dir = 'D:/US/'
avoided_songs_file_path = "./damp_avoided_songs.txt"
id_file_path = "./damp_id.txt"
already_converted_path = 'D:/US_already/'
save_path = "./damp_saved/"

## Check with a song already converted from DALI

The id used should be the same, and "DAMP" annotation added to the source list in id.txt.

In [15]:
main.process_damp_metadata_file(fr_file, fr_dir, DAMP_dir, save_path, already_converted_path, id_file_path, avoided_songs_file_path)

DETECTED FRONT INDEX: 3
DETECTED BACK INDEX 69
Searching for "Diamonds" by Rihanna...
Done.
Searching for "Diamonds" by Rihanna...
Done.


## Check with a song not in DALI

The id used should be a brand new one, in the continuity of the last id generated to convert the last dali file.

In [3]:
main.process_damp_metadata_file(es_file, es_dir, DAMP_dir, save_path, already_converted_path, id_file_path, avoided_songs_file_path)

DETECTED FRONT INDEX: 3
DETECTED BACK INDEX 69
Searching for "Diamonds" by Rihanna...
Done.
Searching for "Diamonds" by Rihanna...
Done.
{'line': 'Find light in the beautiful sea', 'time_index': [11.189583333333333, 13.35], 'time_duration': 2.260416666666666}
1
('Verse', 1) {'content': "Find light in the beautiful sea, I choose to be happy You and I, you and I We're like diamonds in the sky You're a shooting star I see, a vision of ecstasy When you hold me, I'm alive We're like diamonds in the sky", 'singer': ['Rihanna']}
1.0
Before:  None
After:  I choose to be happy You and I, you and I We're like diamonds in the sky You're a shooting star I see, a vision of ecstasy When you hold me, I'm alive We're like diamonds in the sky
{'line': 'I choose to be happy', 'time_index': [13.45, 15.869791666666666], 'time_duration': 2.5197916666666664}
Before:  I choose to be happy You and I, you and I We're like diamonds in the sky You're a shooting star I see, a vision of ecstasy When you hold me, I

## Call for the bulk function

### Real pathes

In [2]:
DAMP_dir = 'D:/US/'
avoided_songs_file_path = "./damp_avoided_songs.txt"
id_file_path = "./damp_id.txt"
already_converted_path = 'D:/US_already/'
save_path = "./damp_saved/"

In [ ]:
main.create_damp_notations(DAMP_dir, save_path, already_converted_path, id_file_path, avoided_songs_file_path)

## Statistics about the DAMP dataset

In [ ]:
utils.count_sources(id_file_path)

In [ ]:
utils.calculate_percentage(id_file_path, already_converted_path, 'DAMP')

In [ ]:
utils.damp_count_avoided_songs(avoided_songs_file_path)

# Post-conversion steps

## Single paragraphs

Is there some songs that got wrongly converted (because of errors in the dataset content that prevent from right line matching, for example) and hence need to be put aside? Usually, these songs only have one single paragraph.

In [3]:
song_ids = utils.get_single_paragraph_song_info("./saved/")

In [4]:
len(song_ids)

304

There are 304 songs we need to put aside. Let's use their id to perform this.

In [5]:
print(song_ids)

{'00000003': ('Geschwisterliebe', 'Die Ärzte'), '00000011': ('Adios', 'Rammstein'), '00000015': ('Kinky Afro', 'Happy Mondays'), '00000048': ('Bad Moon Rising', 'Creedence Clearwater Revival'), '0000006F': ('Hounds Of Love', 'Kate Bush'), '00000082': ('My Love', 'Westlife'), '00000085': ('Have You Ever Seen The Rain', 'Creedence Clearwater Revival'), '0000008A': ('Sweetest Thing', 'U2'), '00000091': ('The Invisible Man', 'Queen'), '0000009C': ('Mine', 'Taylor Swift'), '000000A5': ('Totale Finsternis', 'Tanz Der Vampire'), '000000A8': ('A Boy Named Sue', 'Johnny Cash'), '000000B5': ('Mirror Mirror', 'Blind Guardian'), '000000BC': ('Eet', 'Regina Spektor'), '000000BD': ('Freiheit', 'Marius Müller-Westernhagen'), '000000D0': ('Stray Heart', 'Green Day'), '000000D7': ('Hard Candy Christmas', 'Dolly Parton'), '000000DF': ('Last Man Standing', 'HammerFall'), '000000EB': ('Rosas', 'La Oreja De Van Gogh'), '000000FA': ('Du Entschuldige I Kenn Di', 'Peter Cornelius'), '00000106': ("'39", 'Queen

3364 songs have been converted and are considered as usable. We need to remove the 304 songs from them.

In [ ]:
utils.delete_dict_entries(song_ids, './dali_id.txt')

The 304 songs have been deleted from the id file. Now, let's update the id of the usable songs so that they are all consecutive.

In [6]:
utils.delete_wrong_converted_songs(song_ids, './saved/')

In [ ]:
utils.change_ids_and_rename_files('./dali_id.txt', './saved/', 8)